content/en/docs/architecture/timeouts.md

```diff
@@ -0,0 +1,92 @@
+---
+title: "Job Execution Timeouts"
+date: 2020-12-21T23:27:39-04:00
+draft: false
+---
+
+OpenShift CI/CD platform deploys a [Prow](https://github.com/kubernetes/test-infra/tree/master/prow) instance to manage the lifecycle of jobs. A typical job for Openshift is executed in a Kubernete pod (specified in the job definition and created by Prow) which runs [ci-operator](/docs/architecture/ci-operator/) to assemble an OpenShift release payload, install an ephemeral cluster, run tests against the cluster, and destroy the cluster afterwards. There are timeouts in various phases of a job execution. Some of them are configurable to the developers while others not. In this document, we will explain those timeouts, show the logs if they occur in a job and how to modify their values (if possible).
+
+## Job timeout: 4h by default
+
+The job timeout can be considered as the overall time for a job to run. After that point, Prow will delete the pod for the job.
+
+When a job fails due to the Prow job timeout, the job log contains:
+
+```
+Process did not finish before 4h0m0s timeout
+```
+
+The following stanza in [Prow's configuration](https://github.com/openshift/release/blob/9238ee8b89c861793f9b8e8f7f6509abe33bd0b8/core-services/prow/02_config/_config.yaml#L559) specifies the default value of job timeout and it can be overwritten at `org`, or `org/repo` level (not supported at a more specific level).
+
+```yaml
+plank: # a core component of Prow
+  default_decoration_configs:
+    '*':
+      grace_period: 30m0s
+      timeout: 4h0m0s
+    'org/repo': # overwrite the job timeout at repo level
+      grace_period: 45m0s
+      timeout: 6h0m0s
+```
+
+Any ephemeral cluster on AWS and GCE will be destroyed by [the periodic-ipi-deprovision job](https://github.com/openshift/release/blob/c3a73ae6c1161e242aeaa0943676827161936db8/ci-operator/jobs/infra-periodics.yaml#L377) after 4.5 hours since its creation. This job is to avoid leaking cloud resources in case that jobs fail to clean them up for any reason. **Note** that the `4.5h` here is globe and cannot be overwritten. It is not recommended to modify the job timeout if it impacts a job that requires an ephemeral cluster on those cloud providers.
```

Comment:
This is not really related to this doc. I don't think we need to have it in here.